libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Security/SFUserAccountManager.m

```diff
@@ -457,67 +228,46 @@ - (NSSet*)allExistingAccountNames {
 
 /** Returns a unique user account identifier
  */
-- (NSString*)uniqueUserAccountIdentifier {
+- (NSString*)uniqueUserAccountIdentifier:(NSString *)clientId {
     NSSet *existingAccountNames = [self allExistingAccountNames];
-   
+
     // Make sure to build a unique identifier
     NSString *identifier = nil;
     while (nil == identifier || [existingAccountNames containsObject:identifier]) {
         u_int32_t randomNumber = arc4random();
-        identifier = [NSString stringWithFormat:@"%@-%u", self.oauthClientId, randomNumber];
+        identifier = [NSString stringWithFormat:@"%@-%u", clientId, randomNumber];
     }
 
     return identifier;
 }
 
-- (SFUserAccount*)createUserAccount {
-    SFUserAccount *newAcct = [[SFUserAccount alloc] initWithIdentifier:[self uniqueUserAccountIdentifier]];
+
+- (SFUserAccount*)createUserAccount:(SFOAuthCredentials *)credentials {
+
+    SFUserAccount *newAcct = [[SFUserAccount alloc] initWithIdentifier:[self uniqueUserAccountIdentifier:credentials.clientId] clientId:credentials.clientId];
     SFOAuthCredentials *creds = newAcct.credentials;
```

Comment:
None of the credentials-setting code here and below should remain.  The credentials should already be configured. The user account manager should not be overriding those values.